agent/lm_agent/reconciliation.py

```diff
@@ -144,55 +145,146 @@ async def filter_cluster_update_licenses(licenses_to_update: List) -> List:
 
-    for license_data in filtered_licenses:
+    # Calculate how many licenses should be reserved for each license
+    for license_data in licenses_to_update:
+        # Get license usage from backend
         product_feature = license_data["product_feature"]
-        bookings_sum = license_data["bookings_sum"]
-        license_total = license_data["license_total"]
-        license_used = license_data["license_used"]
-        config_id = await get_config_id_from_backend(product_feature)
-        minimum_value = 0
-        server_type = ""
-        (_, feature) = product_feature.split(".")
-        for config in configs:
-            if config.id == config_id:
-                # Using the total amount of licenses as the minimum value
-                try:
-                    minimum_value = config.features[feature].get("total")
-                    LicenseManagerFeatureConfigurationIncorrect.require_condition(
-                        minimum_value,
-                        f"The configuration for {feature} is incorrect. Please include the total amount of licenses.",
-                    )
-                except AttributeError:
-                    # Fallback to get the total from the old feature format
-                    minimum_value = config.features[feature]
-                server_type = config.license_server_type
-                break
-        slurm_used = await get_tokens_for_license(product_feature + "@" + server_type, "Used")
-        if slurm_used is None:
-            slurm_used = 0
-        new_quantity = license_total - license_used - bookings_sum + slurm_used
-        if new_quantity > license_total:
-            new_quantity = license_total
-        if new_quantity < minimum_value:
-            new_quantity = minimum_value
         product, feature = product_feature.split(".")
-        update_resource = await sacctmgr_modify_resource(product, feature, new_quantity)
-        if update_resource:
-            logger.info("Slurmdbd updated successfully.")
-        else:
-            logger.info("Slurmdbd update unsuccessful.")
-    return r
+        server_used = license_data["license_used"]
+        cluster_name = await get_cluster_name()
+        cluster_booking_sum = await get_booking_sum_for_cluster(cluster_name)
+        other_cluster_booking_sum = await get_booking_sum_for_other_clusters(cluster_name)
+
+        # Get license configuration from backend
+        config_id = await get_config_id_from_backend(product_feature)
+        config = await backend_client.get(f"/lm/api/v1/config/{config_id}")
+        config = config.json()
+
+        license_server_type = config["license_server_type"]
+        # Use feature name to get total and limit from feature data in the license config
+        try:
+            # Get total from new feature format
+            total = config["features"][feature].get("total")
+            LicenseManagerFeatureConfigurationIncorrect.require_condition(
+                total,
```

Comment:
this condition will be raised if the total is 0. I'm assuming that it shouldn't ever be zero, but the message could be confusing in that case.

Reply:
The exception should be raised if the key "total" doesn't exist in the feature's dict.
I'll update the .get("total") to return 0 in that case, which will make this the expected condition for the exception validation.